### PR TITLE
fix: drain completed page_service connections

### PR DESCRIPTION
**Pull request by koivunej**
We've noticed increased memory usage with the latest release. Drain the joinset of `page_service` connection handlers to avoid leaking them until shutdown. An alternative would be to use a TaskTracker. TaskTracker was not discussed in original PR #8339 review, so not hot fixing it in here either.

### CODE REVIEW

**Comment by conradludgate** [comment]
I feel like you could maybe get away with using the lighter weight TaskTracker API which does not need manual draining

**Comment by koivunej** [comment]
> I feel like you could maybe get away with using the lighter weight TaskTracker API which does not need manual draining

Yep, almost wrote it in as a TODO, but I would rather not do that change because I don't really have time to digest all of https://github.com/neondatabase/neon/pull/8339 right now.

**Review by conradludgate — APPROVED** [review]

File: test_runner/regress/test_bad_connection.py [hunk 1]
```diff
@@ -10,7 +10,13 @@
 @pytest.mark.timeout(600)
 def test_compute_pageserver_connection_stress(neon_env_builder: NeonEnvBuilder):
     env = neon_env_builder.init_start()
-    env.pageserver.allowed_errors.append(".*simulated connection error.*")
+    # FIXME currently all errors are logged uniformly
+    # env.pageserver.allowed_errors.append(".*simulated connection error.*") # this is never hit
```
Comment (arpad-m) [inline test_runner/regress/test_bad_connection.py:14]:
```suggestion
    env.pageserver.allowed_errors.append(".*simulated connection error.*") # this is never hit
```

File: test_runner/regress/test_bad_connection.py [hunk 2]
```diff
@@ -10,7 +10,13 @@
 @pytest.mark.timeout(600)
 def test_compute_pageserver_connection_stress(neon_env_builder: NeonEnvBuilder):
     env = neon_env_builder.init_start()
-    env.pageserver.allowed_errors.append(".*simulated connection error.*")
+    # FIXME currently all errors are logged uniformly
```
Comment (arpad-m) [inline test_runner/regress/test_bad_connection.py:13]:
```suggestion
```

**Pull request merged**